inspirehep/modules/records/utils.py

```diff
@@ -430,3 +430,43 @@ def populate_author_suggest(json, *args, **kwargs):
             'input': input_values
         },
     })
+
+
+def get_references_symmetric_differences(ref_list1, ref_list2):
+    """Return the non-common references for the given lists.
+
+    References without the `record` field are just ignored.
+
+    Args:
+        ref_list1 (list): list of references.
+        ref_list2 (list): list of references.
+
+    Returns:
+        list: the symmetric difference of the lists.
+    """
+    list1 = [ref for ref in ref_list1 if ref.get('record')]
+    list2 = [ref for ref in ref_list2 if ref.get('record')]
+
+    list1.sort(key=lambda r: get_recid_from_ref(r['record']))
+    list2.sort(key=lambda r: get_recid_from_ref(r['record']))
+
+    if not ref_list1 or not ref_list2:
+        return list1 or list2
+
+    diff = []
+
+    while list1 and list2:
+        item1 = get_recid_from_ref(list1[0]['record'])
+        item2 = get_recid_from_ref(list2[0]['record'])
+
+        if item1 < item2:
+            diff.append(list1.pop(0))
+        elif item1 > item2:
+            diff.append(list2.pop(0))
+        else:
+            list1.pop(0)
+            list2.pop(0)
+
+    diff.extend(list1)
+    diff.extend(list2)
+    return diff
```

Comment:
maybe something like ``set(a) - set(b)`` for the differences is enough?

Reply:
Nope, Set difference is different from symmetric difference, which is what we want.

https://en.wikipedia.org/wiki/Symmetric_difference

Reply:
I think you want to use `set.symmetric_difference` but maybe you'll have a problem with non hashable types.  In that case you might convert to string.
Also keep in mind the bad performance of deleting from the head with Python lists (they are dynamic arrays that need to be shifted in that case).

Reply:
done